### PR TITLE
mention security changes in upgrade guide, reduce in-line code.

### PR DESCRIPTION
**Pull request by haijian-vaadin**
Mention the (breaking) changes for Fusion Spring Security configurations.
Move inline code to a separate Java file.

### CODE REVIEW

File: articles/guide/upgrading.asciidoc [hunk 1]
```diff
@@ -82,6 +82,17 @@ Modify this behavior with your own followup configuration as needed.
 It also bypasses framework internal and static resources (`/VAADIN/**`, `sw.js` ...).
 Previously, these had to be explicitly matched and ignored in the app.
 
+`VaadinWebSecurityConfigurerAdapter` also configures Spring CSRF token for login and Fusion endpoint calls, so you no longer need to ignore Spring CSRF protection for them like before with `http.csrf().ignoringAntMatchers("/login", "/connect/**");`
```
Comment (joheriks) [inline articles/guide/upgrading.asciidoc:85]:
```suggestion
`VaadinWebSecurityConfigurerAdapter` also configures Spring CSRF token for login and Fusion endpoint requests, so you no longer need to ignore Spring CSRF protection for them like before with `http.csrf().ignoringAntMatchers("/login", "/connect/**");`
```

File: src/main/java/com/vaadin/demo/SecurityConfig.java [hunk 2]
```diff
@@ -1,16 +1,37 @@
...
+    // Configure users and roles in memory
+    auth.inMemoryAuthentication().withUser("user").password("{noop}password").roles("USER");
+  }
+
+  // tag::public-resources[]
+  @Override
+  public void configure(WebSecurity web) throws Exception {
+      super.configure(web);
+      web.ignoring().antMatchers("/icons/**"); 
+      web.ignoring().antMatchers("/images/**");
```
Comment (joheriks) [inline src/main/java/com/vaadin/demo/SecurityConfig.java:34]:
```suggestion
      web.ignoring().antMatchers("/icons/**", "/images/**");
```

File: src/main/java/com/vaadin/demo/SecurityConfig.java [hunk 3]
```diff
@@ -1,16 +1,37 @@
...
+  protected void configure(AuthenticationManagerBuilder auth) throws Exception {
+    // Configure users and roles in memory
+    auth.inMemoryAuthentication().withUser("user").password("{noop}password").roles("USER");
+  }
+
+  // tag::public-resources[]
+  @Override
+  public void configure(WebSecurity web) throws Exception {
+      super.configure(web);
+      web.ignoring().antMatchers("/icons/**");
```
Comment (Artur-) [inline src/main/java/com/vaadin/demo/SecurityConfig.java:33]: Is there a need to have `icons` here in additiopn to `images` as Vaadin automatically makes the icons used in the web manifest available?

**Review by joheriks — CHANGES_REQUESTED** [review]

File: src/main/java/com/vaadin/demo/SecurityConfig.java (hunk 3 above)
Comment (haijian-vaadin) [reply src/main/java/com/vaadin/demo/SecurityConfig.java]: It was copy-pasted code, now removed.

File: src/main/java/com/vaadin/demo/SecurityConfig.java (hunk 2 above)
Comment (haijian-vaadin) [reply src/main/java/com/vaadin/demo/SecurityConfig.java]: removed ignores for icons.

**Review by joheriks — APPROVED** [review]

**Pull request merged**